core/spec/models/spree/order_mutex_spec.rb

```diff
@@ -38,6 +36,23 @@
     end
   end
 
+  context "when another thread requests lock on already locked order" do
+    it "raises a LockFailed error" do
+      Thread.new do
+        Spree::OrderMutex.with_lock!(order) {|o| sleep(3) }
```

Comment:
@philbirt I know you already called out that the specs could use some improvement, but we'll find a better way to do this other than sleeping right?  I'm no threading expert either but happy to pair up and brainstorm ideas if that's helpful. Just lmk.